wgpu-rs/src/lib.rs

```diff
@@ -501,9 +503,17 @@ impl Device {
         let type_size = std::mem::size_of::<T>() as u32;
         assert_ne!(type_size, 0);
 
-        let desc = BufferDescriptor {
-            size: type_size * count as u32,
-            usage,
+        let desc = if count == 0 {
```

Comment:
I think we should be handling this at a lower level: don't pretend the buffer has 1 element. Instead, ensure that the actual raw length of the buffer is not zero (but 1 byte instead) and everywhere else consider it to be 0 sized.

Reply:
I believe I have already made that change?